3d/splitflap.scad

```diff
@@ -433,16 +434,23 @@ module enclosure_front() {
 
 // holes for 28byj-48 motor, centered around motor shaft
 module motor_mount() {
-    circle(r=28byj48_shaft_radius+motor_slop_radius, center=true, $fn=30);
+    circle(r=28byj48_shaft_radius+motor_slop_radius, $fn=30);
     translate([-28byj48_mount_center_offset, -8]) {
-        circle(r=motor_mount_hole_radius, center=true, $fn=30);
+        circle(r=motor_mount_hole_radius, $fn=30);
     }
     translate([28byj48_mount_center_offset, -8]) {
-        circle(r=motor_mount_hole_radius, center=true, $fn=30);
-    }
-    translate([-28byj48_chassis_radius - motor_hole_slop/2, -28byj48_shaft_offset - motor_backpack_extent - motor_hole_slop/2, 0]) {
-        square([28byj48_chassis_radius*2 + motor_hole_slop, 28byj48_chassis_radius + motor_backpack_extent + motor_hole_slop]);
+        circle(r=motor_mount_hole_radius, $fn=30);
     }
+	
+	hull() {
+		x = -28byj48_chassis_radius - motor_hole_slop/2 + motor_window_radius;
+		y = [-28byj48_shaft_offset - motor_backpack_extent - motor_hole_slop/2 + motor_window_radius,
+			-28byj48_shaft_offset + 28byj48_chassis_radius + motor_hole_slop/2 - motor_window_radius];
+		
+		for (i = [0 : 3]) {
+			translate([x * (i%3 ? -1 : 1), y[i%2], 0]) circle(r=motor_window_radius, $fn=40);
+		}
```

Comment:
nit: while this does avoid some repetition, it's only 1 line shorter and arguably harder to understand with 2 different modular arithmetic operations. I would personally probably lean toward just repeating the translate/circle statements 4 times for clarity:
```
        translate([x, y[0], 0]) circle(r=motor_window_radius, $fn=40);
        translate([-x, y[1], 0]) circle(r=motor_window_radius, $fn=40);
        translate([-x, y[0], 0]) circle(r=motor_window_radius, $fn=40);
        translate([x, y[1], 0]) circle(r=motor_window_radius, $fn=40);
```

Reply:
You're absolutely right, writing out all four statements does make it more readable. I'll make that change now. :smiley: